### ISSUE TITLE
Content-type incorrectly parsed by play.api.test.ResultExtractors

### ISSUE THREAD

**Issue by Oduig** [playframework/playframework#4719]
Referring to this line: https://github.com/playframework/playframework/blob/2.3.x/framework/src/play-test/src/main/scala/play/api/test/Helpers.scala#L340

I was using the content type `"text/html;charset=utf-8"`. The string matches the requirement on line 340 but then fails to parse because there is no space after the semicolon. Rather than a `Some("utf-8")` or a `None`, the function returns a `Some("")`. This causes an error somewhere down the line which is quite difficult to trace.

The spec for content-type/charset does not require a space after the semicolon. I would suggest something like the following:
```scala
header(CONTENT_TYPE, of) match {
    case Some(s) => 
        val parts = s.split(";\\s*charset=")
        if (parts.size > 1) Some(parts.drop(1).mkString.trim) else None
    case _ => None
}
```

**Comment by marcospereira** [comment]
Thanks @Oduig! Do you mind submitting a pull requests that fixes this problem? If it comes with some tests/specs it will be even better.

**Comment by marcospereira** [comment]
Fixed by #4723.
